SU2_CFD/src/solvers/CEulerSolver.cpp

```diff
@@ -3122,28 +3125,60 @@ void CEulerSolver::Upwind_Residual(CGeometry *geometry, CSolver **solver_contain
 
       for (iVar = 0; iVar < nPrimVarGrad; iVar++) {
 
-        su2double Project_Grad_i = 0.0;
-        su2double Project_Grad_j = 0.0;
+        const su2double V_ij = V_j[iVar] - V_i[iVar];
```

Comment:
V_ij is not used by the point limiters, it should not be computed by default.

Reply:
It might not used by the limiters, but it's used by the generalized MUSCL scheme whenever Kappa != 0

Reply:
Doh, missed that sorry, will that require some central blending of the flux Jacobians?

Reply:
I guess so, because now we're feeding V_i + 0.5*k*Lim_i*(V_j - V_i) (and similarly for the right state) into CNumerics.

Should I just pass (k*Lim_i, k*Lim_j) into CNumerics and manipulate the Jacobian there?

Reply:
It would be better to pass Lim_i/j and let the numerics read k from the config.

But we should test if the central blending is beneficial, as it will make the system matrix less diagonally dominant.
Perhaps for testing only k needs to be considered (lim_i/j assumed 1), I can help testing if you hack this into Roe (I have a score to settle with some cases that did not work very well with upwind schemes)

Reply:
I added a function `CUpwRoeBase_Flow::GetMUSCLJac()` that multiplies the Jacobians (dF/dU) by (dU/dV)\*(dV_f/dV_i)\* (dV/dU) where dV_f/dV_i is diag(1+0.5\*k\*(Lim_j-Lim_i)), and dU/dV and dV/dU are taken from Hirsch
<img width="552" alt="Screen Shot 2020-07-22 at 1 31 24 PM" src="https://user-images.githubusercontent.com/19416354/88225810-f84bef80-cc1f-11ea-81e3-d08e76425012.png">
<img width="765" alt="Screen Shot 2020-07-22 at 1 31 32 PM" src="https://user-images.githubusercontent.com/19416354/88225819-faae4980-cc1f-11ea-8023-1f4d804b5c1c.png">
I feel like the implementation isn't technically correct because the dV/dU should be evaluated at the nodal values, but from my very limited testing it doesn't seem to break anything (other than like half the regressions)

Reply:
Cool, I will try it, I'm curious about the impact of the more accurate Jacobian vs a possibly simpler approximation that neglects the limiters or uses an average over all variables (which would then boil down to just scaling dF/dU and adding part of the "other" dF/dU and v.v.).
dV/dU are available in `CUpwAUSMPLUS_SLAU_Base_Flow::AccurateJacobian` (btw for those schemes the accurate Jacobian sometimes is worse, hence my curiosity) so if this ends up being the way forward we could factor them out.

Reply:
> other than like half the regressions

I have a script for that :smiling_imp: